dracclient/resources/raid.py

```diff
@@ -98,15 +98,229 @@
 NO_FOREIGN_DRIVE = "STOR018"
 
 
+class RAIDAttribute(object):
+    """Generic RAID attribute class"""
+
+    def __init__(self, name, instance_id, pending_value,
+                 read_only, fqdd):
+        """Creates RAIDAttribute object
+
+        :param name: name of the RAID attribute
+        :param instance_id: InstanceID of the RAID attribute
+        :param pending_value: pending value of the RAID attribute, reflecting
+                an unprocessed change (eg. config job not completed)
+        :param read_only: indicates whether this RAID attribute can be changed
+        :param fqdd: Fully Qualified Device Description of the RAID Attribute
+        """
+
+        self.name = name
+        self.instance_id = instance_id
+        self.pending_value = pending_value
+        self.read_only = read_only
+        self.fqdd = fqdd
+
+    def __eq__(self, other):
+        return self.__dict__ == other.__dict__
+
+    @classmethod
+    def parse(cls, namespace, raid_attr_xml):
+        """Parses XML and creates RAIDAttribute object"""
+
+        name = utils.get_wsman_resource_attr(
+            raid_attr_xml, namespace, 'AttributeName')
+        instance_id = utils.get_wsman_resource_attr(
+            raid_attr_xml, namespace, 'InstanceID')
+        pending_value = utils.get_wsman_resource_attr(
+            raid_attr_xml, namespace, 'PendingValue', nullable=True)
+        read_only = utils.get_wsman_resource_attr(
+            raid_attr_xml, namespace, 'IsReadOnly')
+        fqdd = utils.get_wsman_resource_attr(
+            raid_attr_xml, namespace, 'FQDD')
+
+        return cls(name, instance_id, pending_value,
+                   (read_only == 'true'), fqdd)
+
+
+class RAIDEnumerableAttribute(RAIDAttribute):
+    """Enumerable RAID attribute class"""
+
+    namespace = uris.DCIM_RAIDEnumeration
+
+    def __init__(self, name, instance_id, pending_value,
+                 read_only, fqdd, current_value, possible_values):
+        """Creates RAIDEnumerableAttribute object
+
+        :param name: name of the RAID attribute
+        :param instance_id: InstanceID of the RAID attribute
+        :param pending_value: pending value of the RAID attribute, reflecting
+                an unprocessed change (eg. config job not completed)
+        :param read_only: indicates whether this RAID attribute can be changed
+        :param fqdd: Fully Qualified Device Description of the RAID
+                Attribute
+        :param current_value: list containing the current values of the
+                RAID attribute
+        :param possible_values: list containing the allowed values for the RAID
+                attribute
+        """
+        super(RAIDEnumerableAttribute, self).__init__(name, instance_id,
+                                                      pending_value,
+                                                      read_only, fqdd)
+
+        self.current_value = current_value
```

Comment:
Can you move current_value into RAIDAttribute and move the parsing for it there as well?  Please keep the parameters for __init__ in the same order that we have them for BIOSAttribute, etc.